rotkehlchen/chain/evm/decoding/metamask/decoder.py

```diff
@@ -128,12 +125,14 @@ def _decode_swap(self, context: DecoderContext) -> DecodingOutput:
             else:
                 fee_asset = out_event.asset.resolve_to_evm_token()
 
-        # if fee_asset is a token, update only if it's in the Receive event
+        # update the in_event/out_event to adjust their balance with fees
         fee_amount = asset_normalized_value(amount=fee_raw, asset=fee_asset)
-        full_amount = to_amount + fee_amount
         if in_event.asset == fee_asset:
-            in_event.balance.amount = full_amount
-            in_event.notes = f'Receive {full_amount} {fee_asset.symbol} as the result of a {CPT_METAMASK_SWAPS} swap'  # noqa: E501
+            in_event.balance.amount += fee_amount
+            in_event.notes = f'Receive {in_event.balance.amount} {fee_asset.symbol} as the result of a metamask swap'  # noqa: E501
+        if out_event.asset == fee_asset:
```

Comment:
shouldn't this be elif? Can we have both? From what I understand we can't.
```suggestion
        elif out_event.asset == fee_asset:
```